x/leverage/keeper/keeper.go

```diff
@@ -313,19 +244,12 @@ func (k Keeper) Collateralize(ctx sdk.Context, borrowerAddr sdk.AccAddress, uTok
 		return err
 	}
 
-	err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, borrowerAddr, types.ModuleName, sdk.NewCoins(uToken))
-	if err != nil {
-		return err
-	}
-
-	if err := k.checkCollateralLiquidity(ctx, types.ToTokenDenom(uToken.Denom)); err != nil {
-		return err
-	}
-
-	return k.checkCollateralShare(ctx, uToken.Denom)
+	return k.bankKeeper.SendCoinsFromAccountToModule(ctx, borrowerAddr, types.ModuleName, sdk.NewCoins(uToken))
 }
 
 // Decollateralize disables selected uTokens for use as collateral by a single borrower.
+// This function does NOT check that a borrower remains under their borrow limit.
+// That assertion has been moved to MsgServer.
```

Comment:
Since we're refactoring it anyway to check for both prices, I decided to add an additional refactor which will help with flash loans in the future. (It allows us to take multiple steps - i.e. `Supply + Collateralize` or `Borrow + Liquidate` - before the health check.)